pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -46,9 +47,11 @@
 
     private String serviceUrl;
     @JsonIgnore
+    @EqualsAndHashCode.Exclude
     private transient ServiceUrlProvider serviceUrlProvider;
 
     @JsonIgnore
+    @EqualsAndHashCode.Exclude
     private Authentication authentication;
```

Comment:
This seems to be more a problem in the implementation of `Authentication` interface. If we ignore it here, we could have objects with different auth types or credentials that appear to be equals.